### ISSUE TITLE
twister build fails but returns exit code of 0

### ISSUE THREAD

**Issue by cmorganBE** [zephyrproject-rtos/zephyr#52913]
**Describe the bug**

/scripts/twister does not return a non-zero exit code when the project being built fails to compile.

Please also mention any information which could help others to understand
the problem you're facing:
- Zephyr v3.1
- Nucleo h743zi platform target (not sure this matters)

**To Reproduce**
1. clone zephyr
2. cd tests/kernel/fifo
3. edit fifo_api/src/main.c (add gibberish to the file so the compilation will fail)
4. ../../../scripts/twister --build-only -p nucleo_h743zi -W -T fifo_api
5. Build will show as failed
6. echo $? will print out '0' indicating the script was successful.

**Expected behavior**
Step 6: 'echo $?' should report a non-zero exit code as one or more tests have failed or failed to build.

**Impact**
We had failing tests for a few months because no one was looking at the output and the script that was invoking the scripts/twister was receiving a successful (zero value) exit code from twister.

**Logs and console output**
```
$ git diff
diff --git a/tests/kernel/fifo/fifo_api/src/main.c b/tests/kernel/fifo/fifo_api/src/main.c
index 1ccb3c244a..074e98a11a 100644
--- a/tests/kernel/fifo/fifo_api/src/main.c
+++ b/tests/kernel/fifo/fifo_api/src/main.c
@@ -20,6 +20,7 @@
  * @}
  */
 
+asdfsdsasdfas
 #include <ztest.h>
 extern void test_fifo_thread2thread(void);
 extern void test_fifo_thread2isr(void);
```
```
../../../scripts/twister --build-only -p nucleo_h743zi -W -T fifo_api
ZEPHYR_BASE unset, using "/home/cmorgan/projects/sondebase/zephyr"
Renaming output directory to /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out.4
INFO    - Zephyr version: zephyr-v3.1.0
INFO    - JOBS: 8
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out/testplan.json
INFO    - Using 'zephyr' toolchain.
INFO    - 2 test scenarios (2 configurations) selected, 1 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue

ERROR   - nucleo_h743zi             tests/kernel/fifo/fifo_api/kernel.fifo             FAILED: Build failure
ERROR   - see: /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out/nucleo_h743zi/tests/kernel/fifo/fifo_api/kernel.fifo/build.log
INFO    - Total complete:    2/   2  100%  skipped:    1, failed:    0
INFO    - 0 of 2 test configurations passed (0.00%), 1 failed, 1 skipped with 0 warnings in 7.54 seconds
INFO    - 0 test configurations executed on platforms, 1 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out/twister.json
INFO    - Using 'zephyr' toolchain.
INFO    - Writing xunit report /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out/twister.xml...
INFO    - Writing xunit report /home/cmorgan/projects/sondebase/zephyr/tests/kernel/fifo/twister-out/twister_report.xml...
INFO    - Run completed
cmorgan@cmorgan-bec:~/projects/sondebase/zephyr/tests/kernel/fifo$ echo $?
0
```

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain Zephyr SDK 0.14-2
 - Commit SHA or Version used
```
commit 2ddd73feafd3316af2c547c34d6969bea637d5c6 (HEAD, tag: zephyr-v3.1.0, tag: v3.1.0)
Author: Martí Bolívar <marti.bolivar@nordicsemi.no>
Date:   Fri Jun 3 15:33:24 2022 -0700

    release: Zephyr 3.1.0
    
    Update versions for release v3.1.0.
   
    Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>
```

**Additional context**
None

**Comment by cmorganBE** [comment]
This is fixed in 3.2. I'm not sure where or how, since I think I checked when I found the issue, but I also didn't find exactly where the issue was so maybe I missed the change. We just got updated to 3.2 for an internal project that uses twister and has this issue and it failed as expected when the twister build was broken.

**Comment by cmorganBE** [comment]
@kartben was already closed as completed a few months ago, and it was fixed, sure it isn't planned?
